app/models/piece.rb

```diff
@@ -14,7 +14,7 @@ def move_to!(params)
       # Step 2: Check to see if the piece is of the oppposite color.
       if color != enemy_piece[:color]
         # Step 3: Remove piece from the board.
-        enemy_piece.update_attributes(horizontal_position: nil, vertical_position: nil)
+        enemy_piece.destroy
```

Comment:
This makes more sense, but if you do this you'll have to fix code that relies on piece position values going to nil throughout the codebase.... and any tests. If this seems like a lot of work then I would just change this line back... but then you would have to add a condition(s) to models/game.rb that excludes nil value pieces from running through the loop in the check?(color) method.

For example, here's one spot that relies on pieces position becoming nil:
In games_controller.rb:
`  def show
    @game = Game.find_by_id(params[:id])

    data = params[:p_type]

    until data.nil?
      @game.pieces.where(type: 'Pawn', vertical_position: 1).first.try(:update_attributes, type: data)
      @game.pieces.where(type: 'Pawn', vertical_position: 8).first.try(:update_attributes, type: data)
      data = params[:p_type] = nil
      redirect_to game_path(@game)
    end`